include/libtorrent/i2p_stream.hpp

```diff
@@ -427,8 +427,8 @@ struct i2p_stream : proxy_base
 		TORRENT_ASSERT(m_magic == 0x1337);
 		m_state = read_session_create_response;
 		char cmd[400];
-		int size = std::snprintf(cmd, sizeof(cmd), "SESSION CREATE STYLE=STREAM ID=%s DESTINATION=TRANSIENT\n"
-			, m_id);
+		static const char options[] = "inbound.quantity=3 outbound.quantity=3 inbound.length=3 outbound.length=3";
```

Comment:
is it important to configure the inbound and outbound channels here? I would expect that by leaving these options out, there may be a global configuration that takes effect instead.

Reply:
Default values for Java I2P and i2pd are different.
So to have similar behaviour it is better to specify values directly.
https://geti2p.net/en/docs/api/samv3:
_Please specify the tunnel quantities in the SESSION CREATE message to get consistent performance with the Java I2P and i2pd routers, using the options e.g. inbound.length=3 outbound.length=3._

Reply:
I see. is there a good reason to not include these directly in the format string?

Reply:
Keeping those out of the format string has been a choice of mine when i first tried that. The reason is that those should be parameters that should be available to set when creating the connection by the client.

Reply:
Yes, I made it like that because this is how it was suggested by @jigen.
Also I worried about too long strings not fitting in text editor.
It is possible to solve it by breaking string in two parts.
I was thinking about such feature as something non-standard, but I searched more about it, and now I know that adjacent string literals concatenation is perfectly fine method.
I will make this change.

Reply:
I even thought that separating made it more readable than a long string.

Reply:
My recommendations about the options, although perhaps better implemented by arvidn later than by Vort in this PR, are here: https://github.com/arvidn/libtorrent/issues/7262#issuecomment-1385362013

Reply:
@jiigen look how I made it in 022b53cdc9b42884bd1fb2303be8f81cf5758472.
It is separated and at the same time it is single string.

Reply:
Looks fine to me.